test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeUnsafeTests.java

```diff
@@ -190,4 +221,241 @@ static public void testuninitializedVTClassHasNullDefaultValue() {
 		}
 		assertNull(myUnsafe.uninitializedDefaultValue(NeverInitialized.class));
 	}
+
+	@Test
+	static public void testNullObjGetValue() throws Throwable {
+		assertThrows(NullPointerException.class, () -> {
+			myUnsafe.getValue(null, 0, ValueTypeInt.class);
+		});
+	}
+
+	@Test
+	static public void testNullClzGetValue() throws Throwable {
+		assertThrows(NullPointerException.class, () -> {
+			myUnsafe.getValue(vtPoint, vtPointOffsetX, null);
+		});
+	}
+
+	@Test
+	static public void testNonVTClzGetValue() throws Throwable {
+		assertNull(myUnsafe.getValue(vtPoint, vtPointOffsetX, IntWrapper.class));
+	}
+
+	@Test
+	static public void testGetValuesOfArray() throws Throwable {
+		if (isFlatteningEnabled) {
+			ValueTypePoint2D p = myUnsafe.getValue(vtPointAry, vtPointAryOffset0, ValueTypePoint2D.class);
+			assertEquals(p.x.i, vtPointAry[0].x.i);
+			assertEquals(p.y.i, vtPointAry[0].y.i);
+			p = myUnsafe.getValue(vtPointAry, vtPointAryOffset1, ValueTypePoint2D.class);
+			assertEquals(p.x.i, vtPointAry[1].x.i);
+			assertEquals(p.y.i, vtPointAry[1].y.i);
+		}
+	}
+
+	@Test
+	static public void testGetValueOfZeroSizeVTArrayDoesNotCauseError() throws Throwable {
+		ZeroSizeValueType[] zsvtAry = new ZeroSizeValueType[] {
+			new ZeroSizeValueType(),
+			new ZeroSizeValueType()
+		};
+		long zsvtAryOffset0 = myUnsafe.arrayBaseOffset(zsvtAry.getClass());
+		assertNotNull(myUnsafe.getValue(zsvtAry, zsvtAryOffset0, ZeroSizeValueType.class));
+	}
+
+	@Test
+	static public void testGetValueOfZeroSizeVTObjectDoesNotCauseError() throws Throwable {
+		ZeroSizeValueTypeWrapper zsvtw = new ZeroSizeValueTypeWrapper();
+		long zsvtwOffset0 = myUnsafe.objectFieldOffset(ZeroSizeValueTypeWrapper.class.getDeclaredField("z"));
+		assertNotNull(myUnsafe.getValue(zsvtw, zsvtwOffset0, ZeroSizeValueType.class));
+	}
+
+	@Test
+	static public void testGetValuesOfObject() throws Throwable {
+		if (isFlatteningEnabled) {
+			ValueTypeInt x = myUnsafe.getValue(vtPoint, vtPointOffsetX, ValueTypeInt.class);
+			assertEquals(x.i, vtPoint.x.i);
+			ValueTypeInt y = myUnsafe.getValue(vtPoint, vtPointOffsetY, ValueTypeInt.class);
+			assertEquals(y.i, vtPoint.y.i);
+		}
+	}
+
+	@Test
+	static public void testGetValueOnVTWithLongFields() throws Throwable {
+		if (isFlatteningEnabled) {
+			ValueTypeLongPoint2D vtLongPoint = new ValueTypeLongPoint2D(123, 456);
+			ValueTypeLong x = myUnsafe.getValue(vtLongPoint, vtLongPointOffsetX, ValueTypeLong.class);
+			assertEquals(x.l, vtLongPoint.x.l);
```

Comment:
Please also add an assert that `x.l` equals the actual value 123. Similarly for other tests here.

Reply:
I just want to make sure `x.l == vtLongPoint.x.l == 123`

Reply:
addressed in c83a184